pkg/kubelet/prober/worker.go

```diff
@@ -46,6 +45,9 @@ type worker struct {
 	// Describes the probe configuration (read-only)
 	spec *api.Probe
 
+	// The type of the worker.
+	probeType probeType
+
```

Comment:
Done (mostly - probeType is still passed to prober, I'll try to fix this in a later PR). Also made a few cleanup changes:

- prober.Manager is a worker field rather than method parameter
- removed side effect (`go w.run()`) from worker constructor for easier testing. This also exposed a bug in TestCleanUp which I fixed.